pkg/common/namespace_selection.go

```diff
@@ -289,26 +311,55 @@ func (r *NamespaceSelectorReconciler) Get(name string, t policyv1.Target) ([]str
 
 // HasUpdate indicates when the cached selection for this policy has been changed since the last
 // time that Get was called for that policy.
-func (r *NamespaceSelectorReconciler) HasUpdate(name string) bool {
+func (r *NamespaceSelectorReconciler) HasUpdate(namespace string, name string) bool {
 	r.lock.RLock()
 	defer r.lock.RUnlock()
 
-	return r.selections[name].hasUpdate
+	return r.selections[getKey(namespace, name)].hasUpdate
 }
 
 // Stop tells the SelectorReconciler to stop updating the cached selection for the name.
-func (r *NamespaceSelectorReconciler) Stop(name string) {
+func (r *NamespaceSelectorReconciler) Stop(namespace string, name string) {
 	r.lock.Lock()
 	defer r.lock.Unlock()
 
-	delete(r.selections, name)
+	delete(r.selections, getKey(namespace, name))
+}
+
+func getKey(namespace string, name string) string {
+	return fmt.Sprintf("%s/%s", namespace, name)
 }
 
-func (r *NamespaceSelectorReconciler) update(name string, sel namespaceSelection) {
+func splitKey(key string) (string, string) {
```

Comment:
Is this return (namespace, name).. ? Can we write return value in function description (comment)